lib/grpc/stream.ex

```diff
@@ -139,121 +115,89 @@ defmodule GRPC.Stream do
 
       flow = GRPCStream.single(request, max_demand: 5)
   """
-  @spec single(any(), Keyword.t()) :: t()
-  def single(input, opts \\ []) when is_struct(input),
+  @spec unary(any(), Keyword.t()) :: t()
+  def unary(input, opts \\ []) when is_struct(input),
     do: build_grpc_stream([input], Keyword.merge(opts, unary: true))
 
   @doc """
-  Converts a single gRPC request into a `Flow` pipeline while extracting and propagating context metadata from the `GRPC.Server.Stream`.
-
-  This is useful for unary gRPC calls that require metadata propagation within a Flow-based pipeline.
-
-  ## Parameters
-
-    - `input`: The enumerable stream of incoming gRPC messages.
-    - `materializer`: A `%GRPC.Server.Stream{}` struct representing the current gRPC stream context.
-    - `opts`: Optional keyword list for configuring the Flow or GenStage producer.
-
-  ## Behavior
+  Extracts the internal `Flow` pipeline from a `GRPC.Stream`.
 
-    - Automatically extracts gRPC metadata from the stream headers and injects it into the options as `:metadata`.
+  Raises an `ArgumentError` if the `flow` has not been initialized.
 
   ## Returns
 
-    - A `GRPC.Stream` that emits the single gRPC message under demand.
-
-  ## Example
-
-      flow = GRPC.Stream.single_as_ctx(request, stream, max_demand: 10)
-
+    The internal `Flow` struct.
   """
-  @spec single_as_ctx(any(), GRPC.Server.Stream.t(), Keyword.t()) :: t()
-  def single_as_ctx(input, %GRPC.Server.Stream{} = materializer, opts \\ [])
-      when is_struct(input) do
-    meta = get_headers(materializer) || %{}
-    single(input, Keyword.merge(opts, metadata: meta))
-  end
+  @spec to_flow(t()) :: Flow.t()
+  def to_flow(%__MODULE__{flow: flow}) when is_nil(flow), do: Flow.from_enumerable([])
 
-  @doc """
-  Wraps an existing `Flow` into a `GRPC.Stream` struct.
+  def to_flow(%__MODULE__{flow: flow}), do: flow
 
-  > Note: `:join_with` is not supported in this mode and will raise an error if passed.
+  @doc """
+  Executes the flow for unary/single streams and emits responses into the provided gRPC server stream.
 
   ## Parameters
 
-    - `flow`: A valid `Flow` pipeline.
-    - `opts`: Optional keyword options.
+    - `flow`: A `GRPC.Stream` struct containing the flow to be executed.
+    - `stream`: A `GRPC.Server.Stream` to which responses are sent.
+    - `:dry_run` — If `true`, responses are not sent (used for testing or inspection).
```

Comment:
Not sure we should introduce options just for testing. Is this really useful for users?

Reply:
I think so because it allows users to test their own workflows.